pkgs/development/misc/resholve/resholve-utils.nix

```diff
@@ -185,6 +185,15 @@ rec {
       # retain a reference to the base
       passthru = unresholved.passthru // {
         unresholved = unresholved;
+        /*
+        update bot wants to find outputHash of the src but
+        we use inner derivation as the src. (update bot is
+        looking at outer.src.outputHash and the value is at
+        outer.src.src.outputHash). Abusing passthru to tell
+        a white lie about our src (claiming our src is
+        inner.src).
+        */
+        src = unresholved.src;
```

Comment:
This looks really problematic. For one, I often use `.src` to investigate package issues, and overwriting it here would probably cause a few head scratches.

What do wrapped packages do to workaround this problem? For example, `mpv`?

Reply:
It doesn't, AFAICT:

```console
$ nix eval -f . --raw pkgs.mpv.src.drvAttrs.outputHash
error: attribute 'src' in selection path 'pkgs.mpv.src.drvAttrs.outputHash' not found
```

Reply:
This is not what I meant (kinda expect that they don't overwrite the `src).

What I meant is: they still get automatic updates, right? If yes, we should copy whatever they're doing.

Reply:
Right--but best I can tell these are two sides of the same coin. The log cited in the OP shows pretty clearly that this is where the bot falls over.

I see no r-ryantm update PRs for mpv. All of the bump commits on the first few pages of the commit history for the file (https://github.com/NixOS/nixpkgs/commits/f2ebc057eeddfe8bb207fdadc8d78ae1976d43ee/pkgs/applications/video/mpv/default.nix)

I imagine his notifications are too busy to notice this, but just in case: perhaps @ryantm has thoughts on how/whether nixpkgs vs nixpkgs-update should be accommodating this?

Reply:
> I see no r-ryantm update PRs for mpv. All of the bump commits on the first few pages of the commit history for the file (https://github.com/NixOS/nixpkgs/commits/f2ebc057eeddfe8bb207fdadc8d78ae1976d43ee/pkgs/applications/video/mpv/default.nix) are made by humans.

Ok, this is fair. Now that I think about it, what happens in most wrapped packages is something like this:

```
❯ rg buildah
pkgs/top-level/all-packages.nix
3122:  buildah = callPackage ../development/tools/buildah/wrapper.nix { };
3123:  buildah-unwrapped = callPackage ../development/tools/buildah { };
```

And of course [auto-updates works](https://github.com/NixOS/nixpkgs/pull/190779) because the `buildah-unwrapped` is an actual derivation.

> Right--but best I can tell these are two sides of the same coin. The log cited in the OP shows pretty clearly that this is where the bot falls over.

I still don't like the current solution. Aside from the debugging, it can cause all sort of other issues since people expect `src == current derivation src` instead of `src == super derivation src`.

Would be better to change this behavior in @r-ryantm/nixpkgs-update itself.